### ISSUE TITLE
Appears to be an issue with some classes on iOS 10

### ISSUE THREAD

**Issue by NathanaelA** [NativeScript/ns-v8ios-runtime#67]
Basically, the first time the UILayoutGuide.alloc().init() is everything is created correctly, and the object works as expected.
The second time thru the routine, the UILayoutGuide.alloc() works; but the prototype chain is now missing the majority of the functions one of which is the `init` function, so then the application crashes with init not being a function.

- Appears to be directly related to the .init() call
- Can duplicate on a iOS 10 Emulator w/ XCode 12
- See https://github.com/NativeScript/NativeScript/pull/8954 for PR used to fix the issue.

Prototype Chain on first `alloc()` (120 items):
```
-> bottomAnchor
-> centerXAnchor
-> centerYAnchor
-> hasAmbiguousLayout
-> heightAnchor
-> identifier
-> layoutFrame
-> leadingAnchor
-> leftAnchor
-> owningView
-> rightAnchor
-> topAnchor
-> trailingAnchor
-> widthAnchor
-> constraintsAffectingLayoutForAxis
-> encodeWithCoder
-> initWithCoder
-> toString
-> accessibilityActivationPoint
-> accessibilityCustomActions
-> accessibilityCustomRotors
-> accessibilityElements
-> accessibilityElementsHidden
-> accessibilityFrame
-> accessibilityHint
-> accessibilityLabel
-> accessibilityLanguage
-> accessibilityNavigationStyle
-> accessibilityPath
-> accessibilityTraits
-> accessibilityValue
-> accessibilityViewIsModal
-> autoContentAccessingProxy
-> classForCoder
-> classForKeyedArchiver
-> isAccessibilityElement
-> observationInfo
-> shouldGroupAccessibilityChildren
-> accessibilityActivate
-> accessibilityAssistiveTechnologyFocusedIdentifiers
-> accessibilityDecrement
-> accessibilityElementAtIndex
-> accessibilityElementCount
-> accessibilityElementDidBecomeFocused
-> accessibilityElementDidLoseFocus
-> accessibilityElementIsFocused
-> accessibilityIncrement
-> accessibilityPerformEscape
-> accessibilityPerformMagicTap
-> accessibilityScroll
-> addObserverForKeyPathOptionsContext
-> awakeAfterUsingCoder
-> awakeFromNib
-> copy
-> dealloc
-> dictionaryWithValuesForKeys
-> didChangeValueForKey
-> didChangeValueForKeyWithSetMutationUsingObjects
-> didChangeValuesAtIndexesForKey
-> doesNotRecognizeSelector
-> finalize
-> forwardInvocation
-> forwardingTargetForSelector
-> indexOfAccessibilityElement
-> init
-> methodForSelector
-> methodSignatureForSelector
-> mutableArrayValueForKey
-> mutableArrayValueForKeyPath
-> mutableCopy
-> mutableOrderedSetValueForKey
-> mutableOrderedSetValueForKeyPath
-> mutableSetValueForKey
-> mutableSetValueForKeyPath
-> observeValueForKeyPathOfObjectChangeContext
-> performSelectorInBackgroundWithObject
-> performSelectorOnMainThreadWithObjectWaitUntilDone
-> performSelectorOnMainThreadWithObjectWaitUntilDoneModes
-> performSelectorOnThreadWithObjectWaitUntilDone
-> performSelectorOnThreadWithObjectWaitUntilDoneModes
-> performSelectorWithObjectAfterDelay
-> performSelectorWithObjectAfterDelayInModes
-> prepareForInterfaceBuilder
-> removeObserverForKeyPath
-> removeObserverForKeyPathContext
-> replacementObjectForCoder
-> replacementObjectForKeyedArchiver
-> setNilValueForKey
-> setValueForKey
-> setValueForKeyPath
-> setValueForUndefinedKey
-> setValuesForKeysWithDictionary
-> validateValueForKeyError
-> validateValueForKeyPathError
-> valueForKey
-> valueForKeyPath
-> valueForUndefinedKey
-> willChangeValueForKey
-> willChangeValueForKeyWithSetMutationUsingObjects
-> willChangeValuesAtIndexesForKey
-> autorelease
-> class
-> conformsToProtocol
-> isEqual
-> isKindOfClass
-> isMemberOfClass
-> performSelector
-> performSelectorWithObject
-> performSelectorWithObjectWithObject
-> release
-> respondsToSelector
-> retain
-> retainCount
-> self
-> debugDescription
-> description
-> hash
-> isProxy
-> superclass
-> zone
```

Prototype Chain on second `alloc()`   (40 items):
```
-> bottomAnchor
-> centerXAnchor
-> centerYAnchor
-> hasAmbiguousLayout
-> heightAnchor
-> identifier
-> layoutFrame
-> leadingAnchor
-> leftAnchor
-> owningView
-> rightAnchor
-> topAnchor
-> trailingAnchor
-> widthAnchor
-> constraintsAffectingLayoutForAxis
-> encodeWithCoder
-> initWithCoder
-> toString
-> length
-> autorelease
-> class
-> conformsToProtocol
-> isEqual
-> isKindOfClass
-> isMemberOfClass
-> performSelector
-> performSelectorWithObject
-> performSelectorWithObjectWithObject
-> release
-> respondsToSelector
-> retain
-> retainCount
-> self
-> debugDescription
-> description
-> hash
-> isProxy
-> superclass
-> zone
```

**Comment by darind** [comment]
Is this behaviour specific to iOS 10 or is it also reproducible on higher iOS versions?

**Comment by NathanaelA** [comment]
Easily repro on iOS 10.  I didn't see this issue on 11+ as "init" always worked -- however I didn't step through the code to see if the prototype chain had any missing items on 11+, it is possible that the chain is missing different items on 11+.     If using XCode 12, just download the iOS 10.3 simulator, it will fail on it if you change back to the .alloc().init() on the second pass through.

**Comment by darind** [comment]
@NathanaelA, could you please provide steps to reproduce this?

Here's what I did:

1. Open the [ios v8 runtime source code](https://github.com/nativeScript/ns-v8ios-runtime/) with XCode 12
2. Replace the contents of [TestRunner/app/index.js](https://github.com/NativeScript/ns-v8ios-runtime/blob/master/TestRunner/app/index.js) with the following:

```javascript
const getKeys = (obj) => {
    let properties = new Set();
    let currentObj = obj;
    do {
        Object.getOwnPropertyNames(currentObj).map(item => properties.add(item));
    } while ((currentObj = Object.getPrototypeOf(currentObj)));
    return [...properties.keys()];
}

const guide1 = UILayoutGuide.alloc().init();
const keys1 = JSON.stringify(getKeys(guide1));

const guide2 = UILayoutGuide.alloc().init();
const keys2 = JSON.stringify(getKeys(guide2));

console.log(keys1 == keys2);

UIApplicationMain(0, null, null, null);
```

3. Run the `TestRunner` target under an iPhone 6s simulator target (running iOS 10.3)

In my case this prints: `true` (meaning that both `UILayoutGuide` instances have been successfully initialized and have the same prototype)

**Comment by NathanaelA** [comment]
Yeah, I didn't run it via the TestRunner, I ran the NS framework as a full `tns create test --js` application just to figure out why it was throwing the error "init()" isn't defined and dropped a debug point on the line 204 of `core/ui/core/view/view-helper/index.ios.ts` where UILayoutGuide is at.

That verified I could duplicate the crash on iOS 10 simulator...

I then modified the code where UILayoutGuide.alloc().init() and broke it into two separate statements in the source and also several console.log's to get me all the keys once I realized that the values were different.   I also played with adding a second one into the same code just to see what would happen.

Also  note the keys should have been from UILayoutGuide().alloc()   -- NOT the variable returned by init().   Because the issue was that what was ALLOCATED was missing its prototype chain values, including the "init" function, 

I'm trying to recall if I saw the issue when I did two of them in a row, or if my test had the second `alloc().init()` after line #206 where it activated the prior constraint...   I did a lot of tests until I realized that the prototype chain was different on the second loop into that function.
